src/backend/distributed/commands/table.c

```diff
@@ -1170,7 +1196,21 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		sqlForTaskList = DeparseTreeNode((Node *) newStmt);
 	}
 
-	ddlJob->metadataSyncCommand = useInitialDDLCommandString ? alterTableCommand : NULL;
+	/* In the case of ADD PRIMARY KEY command,
```

Comment:
So far, we set the `metadataSyncCommand` to NULL if we don't want to send the original DDL command string, and then in PostprocessAlterTableStmt, we take care of the metadata sync command. So far, this is only used for in cases where worker_nextval should be used instead of nextval() in column defaults with type int and smallint.

NOW with this PR, we want to add another subcase to this. But we actually don't need to do this in PostprocessAlterTableStmt, we can do it here. So, I suggest we do something like:
``` C
ddlJob->metadataSyncCommand = NULL;
if(useInitialDDLCommandString)
{
ddlJob->metadataSyncCommand = alterTableCommand;
}
else
{
if(primary key constraint with no name)
{
ddlJob->metadataSyncCommand = sqlForTaskList;
}
}
```

Reply:
or If you think of something more readable than this you can do that